org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java

```diff
@@ -464,18 +484,13 @@ private void handleEvent(DebugEvent debugEvent) {
             for (Events.StoppedEvent stoppedEvent : this.stoppedEventsByThread.values()) {
                 this.sendEvent(stoppedEvent);
             }
-        } else if (event instanceof BreakpointEvent) {
-            BreakpointEvent bpEvent = (BreakpointEvent) event;
-            ThreadReference bpThread = bpEvent.thread();
-            Location bpLocation = bpEvent.location();
+        } else if (event instanceof BreakpointEvent || event instanceof StepEvent) {
             try {
-                Types.Source clientSource = this.convertDebuggerSourceToClient(bpLocation);
-                Events.StoppedEvent stopevent = new Events.StoppedEvent("breakpoint", clientSource,
-                        this.convertDebuggerLineToClient(bpLocation.lineNumber()), 0, "", bpThread.uniqueID());
-                this.stoppedEventsByThread.put(bpThread.uniqueID(), stopevent);
-                this.sendEvent(stopevent);
+                Events.StoppedEvent stopEvent = this.convertDebuggerStoppedEventToClient((LocatableEvent) event);
+                this.stoppedEventsByThread.put(stopEvent.threadId, stopEvent);
```

Comment:
As discussed, we will not maintain internal state of the stopped threads but let VSCode handle the state management. Issue should be created at their side.